evdns.c

```diff
@@ -1687,6 +1687,13 @@ nameserver_read(struct nameserver *ns) {
 			    evutil_socket_error_to_string(err), err);
 			goto done;
 		}
+        if(r == 0 && recv_zero++ > 10) {
```

Comment:
```
       Datagram sockets in various domains (e.g., the UNIX and Internet domains) permit zero-length datagrams.  When such a datagram is received, the return value is 0.
```

But why sends them to that socket?
Can you write a reproducer?

Also patch is not suitable for upstream in this form:
- alignment and coding style should look the same as the code around
- 10 is a magic constant, it should be moved to some const/define
- these code also looks icky, I'm more or less ok if the purpose will be described and it will be made under some option, see `evdns_base_set_option`